pelican/plugins/myst_reader/myst_reader.py

```diff
@@ -50,20 +51,53 @@ class MySTReader(BaseReader):
     file_extensions = FILE_EXTENSIONS
 
     def __init__(self, *args, **kwargs):
+        """Fetch settings from ``pelicanconf.py`` and initialize MyST parser."""
         super().__init__(*args, **kwargs)
 
-        # Get settings set in pelicanconf.py
-        extensions = self.settings.get("MYST_EXTENSIONS", [])
-        self.parser_config = main.MdParserConfig(
-            # renderer="docutils",
-            enable_extensions=extensions
+        # Load MyST user settings.
+        myst_settings = self.settings.get("MYST_SETTINGS", dict())
+        # Get the list of user-activated MyST extensions.
+        # XXX Maybe get rid of MYST_EXTENSIONS entirely since we can set MYST_SETTINGS["enable_extensions"] from pelicanconf.py instead?
+        myst_settings.setdefault(
+            "enable_extensions", self.settings.get("MYST_EXTENSIONS", set())
         )
-        self.md_parser = create_md_parser(self.parser_config, RendererHTML)
-        self.myst_extensions = self.parser_config.enable_extensions
+        # Set the default list of warnings to suppress. List available at:
+        # https://myst-parser.readthedocs.io/en/latest/configuration.html#build-warnings
+        myst_settings.setdefault(
+            "suppress_warnings",
+            [
+                # Ignore the "Document headings start at H2, not H1 [myst.header]" warning,
+                # since Pelican will use the title field in the front-matter as the H1 header of the page.
+                "myst.header",
+            ],
+        )
+        # Parse, validate and normalize MyST settings.
+        self.myst_config = MdParserConfig(**myst_settings)
+
+        # Default Docutils settings.
+        # These are the same default as hard-coded in Pelican:
+        # https://github.com/getpelican/pelican/blob/1f6b344/pelican/readers.py#L255-L262
+        self.docutils_settings = {
+            "initial_header_level": "2",
+            "syntax_highlight": "short",
+            "input_encoding": "utf-8",
+            "halt_level": 2,
+            "traceback": True,
+            "warning_stream": StringIO(),
+            "embed_stylesheet": False,
+        }
+
+        # Transfer MyST settings to Docutils settings.
+        # This is necessary to align the Docutils parser with the MyST parser.
+        for param_id in self.myst_config.as_dict():
+            self.docutils_settings[f"myst_{param_id}"] = getattr(
+                self.myst_config, param_id
+            )
```

Comment:
That part may not be clear enough I guess. Here is what I am trying to do.

The problem is that we use two parsers:
- The MySt parser to produce tokens
- The Docutils parser (or Sphinx) to produce HTML

These 2 steps are independents and each parser has its own way of getting parameters. The standalone MyST parser use the [same IDs as the frontmatter (local) configuration](https://myst-parser.readthedocs.io/en/latest/configuration.html#frontmatter-local-configuration), while the [Docutils parameters needs to be prefixed with `myst_`](https://myst-parser.readthedocs.io/en/latest/configuration.html#global-configuration) to be passed to its own MyST parser instance (like Sphinx).

To have the same behaviour accross the 2 components (+ Sphinx), I set the defaults of `self.docutils_settings` from `self.myst_config`, while prefixing them with `myst_`. Then I update the `self.docutils_settings` with the user's own values from `pelicanconf.py`.

That way a user can sets different configuration for the MySt parser and Docutils parser. But by default everything's aligned and there is no surprises.

This means a user can have in its `pelicanconf.py`:
```python
MYST_SETTINGS = {
    "gfm_only": True,
}

DOCUTILS_SETTINGS = {
    "myst_gfm_only": False,
}
```

Reply:
Now with this explanation, how would you like to have the code change?

Reply:
I studied your implementation a bit. I would like the user facing setttings to be like

```py
MYST_SPHINX_SETTINGS = {
    "myst_gfm_only": True,
}

MYST_DOCUTILS_SETTINGS = {
    "myst_gfm_only": False,
}
```
So they appear cohesive and the dictionary keys can easily be related to. Do you agree?


Also if possible add two unit tests which uses these variables. Right now the test cases work based on `MYST_EXTENSIONS` alone.

Reply:
Alternatively it can also be

> ```python
> MYST_SPHINX_SETTINGS = {
>     "gfm_only": True,
> }
> 
> MYST_DOCUTILS_SETTINGS = {
>     "gfm_only": False,
> }
> ```

Btw, internally I don't think we should mix the two variables. I.e. we should keep the two rendering backends separately configurable. The user can easily do, `MYST_SPHINX_SETTINGS.update(MYST_DOCUTILS_SETTINGS)` in his/her pelicanconf.py.

Reply:
OK!

So, to sum up the requested changes:
- I will add a new `MYST_SPHINX_SETTINGS` alongside `MYST_SETTINGS` and `DOCUTILS_SETTINGS`, that will be exclusively used to feed the Sphinx-based rendering mode
- I will update the documentation to invite users to use the short-form parameter IDs (i.e. `gfm_only` instead of `myst_gfm_only`)
- Make sure to translate the short parameter IDs to the `myst_` prefixed ones for Docutils and Sphinx.

Reply:
Thinking about it, and in light of your first observation, I'd rather not implement the prefixing with `myst_`. This might be too much magic and can add friction in the future if the sub-projects are changing their conventions.

Reply:
Still, what do you think of auto-aligning the settings of the MyST parser and the Docutils parser? Should we get rid of this and invite the user to carefully setup its 3 constants `MYST_SETTINGS`, `MYST_SPHINX_SETTINGS` and `DOCUTILS_SETTINGS`?

I fear this might lower the user-friendliness of the plugin and increase support. Setting up a Pelican blog is still a lot of work, so if we can make this plugin just work it would increase its adoption.

Reply:
No need for the ambiguous `MYST_SETTINGS`. Only MYST_SPHINX_SETTINGS and **MYST_DOCUTILS_SETTINGS**.

Sure in theory we could have delegated a single variable and do all the logic within the plugin. But I would rather have something maintainable and not prone to the changes upstream. Maybe in the future we can also implement a renderer for the Node JS based CLI for MyST. Which would require another variable...

Reply:
In any case when user don't set these variables we can have reasonable defaults to reduce friction. You can choose the defaults! 🙂

Reply:
OK I get it. And I didn't see that JS based CLI coming! 😅

The defaults are as they are now. I'll probably send another PR in a couple of weeks as I play around the iodiosyncracies of all the rendering stack.

Last question: should I get rid of `MYST_EXTENSIONS` too? In theory, nothing prevent the user to do:
```python
MYST_DOCUTILS_SETTINGS = {
    "myst_gfm_only": False,
    "myst_enable_extensions": [
        "amsmath",
        "tasklist",
    ],
}
```

Reply:
OK I just pushed the last round of fixes.